it/v6/ukPropertyBsas/submit/def3/Def3_SubmitUkPropertyBsasISpec.scala

```diff
@@ -0,0 +1,219 @@
+import play.api.http.Status._
+import play.api.libs.json._
+import play.api.libs.ws.{WSRequest, WSResponse}
+import play.api.test.Helpers.AUTHORIZATION
+import shared.models.errors.{BadRequestError, _}
+import shared.models.utils.JsonErrorValidators
+import shared.stubs._
+import support.IntegrationBaseSpec
+import v6.ukPropertyBsas.submit.def3.model.request.SubmitUKPropertyBsasRequestBodyFixtures.fullRequestJson
+
+class Def3_SubmitUkPropertyBsasISpec extends IntegrationBaseSpec with JsonErrorValidators {
+
+  "Calling the Submit UK Property Accounting Adjustments endpoint" should {
+    "return a 200 status code" when {
+
+      "any valid request is made for a TYS tax year" in new TysIfsTest {
+        override def setupStubs(): Unit = {
+          stubDownstreamSuccess()
+        }
+
+        val response: WSResponse = await(request().post(fullRequestJson))
+        response.status shouldBe OK
+        response.header("Content-Type") shouldBe None
+      }
+    }
+
+    "return validation error according to spec" when {
+      "validation error" when {
+        def validationErrorTest(requestNino: String,
+                                requestCalculationId: String,
+                                requestTaxYear: Option[String],
+                                requestBody: JsValue,
+                                expectedStatus: Int,
+                                expectedBody: MtdError): Unit = {
+          s"validation fails with ${expectedBody.code} error" in new TysIfsTest {
+
+            override val nino: String            = requestNino
+            override val calculationId: String   = requestCalculationId
+            override val taxYear: Option[String] = requestTaxYear
+
+            val response: WSResponse = await(request().post(requestBody))
+            response.status shouldBe expectedStatus
+            response.json shouldBe Json.toJson(expectedBody)
+          }
+        }
+
+        val input = List(
+          ("AA1234A", "041f7e4d-87b9-4d4a-a296-3cfbdf92f7e2", Some("2025-26"), fullRequestJson, BAD_REQUEST, NinoFormatError),
+          ("AA123456A", "041f7e4d87b9", Some("2025-26"), fullRequestJson, BAD_REQUEST, CalculationIdFormatError),
+          (
+            "AA123456A",
+            "041f7e4d-87b9-4d4a-a296-3cfbdf92f7e2",
+            Some("2025-26"),
+            fullRequestJson.replaceWithEmptyObject("/income"),
+            BAD_REQUEST,
+            RuleIncorrectOrEmptyBodyError.copy(paths = Some(List("/income")))
+          ),
+          (
+            "AA123456A",
+            "041f7e4d-87b9-4d4a-a296-3cfbdf92f7e2",
+            Some("2025-26"),
+            fullRequestJson.update("/expenses/consolidatedExpenses", JsNumber(1.23)),
+            BAD_REQUEST,
+            RuleBothExpensesError.copy(paths = Some(List("/expenses")))
+          )
+        )
+        input.foreach(args => (validationErrorTest _).tupled(args))
+      }
+
+      "tax year validation errors" when {
```

Comment:
Ok. Interesting. But this means bad tax years will validate differently based on whether they have a payload that happens to align with what we call the 'default' schema. Having a default schema probably makes little sense.

We need our code/functionality to be driven by our tests (rather than the other way around). One for team discussion, I think as this may raise its head elsewhere (or it might have been worked around elsewhere).